## csv 파일 가져와서 데이터 프레임 만들기 ##
pd.read_csv('파일명')

*cf. pd.read_excel('파일명'): 엑셀파일 가져와서 데이터 프레임 만들기*

In [4]:
import pandas as pd

df_midwest = pd.read_csv('midwest.csv')
df_midwest

PID     county state   area  poptotal   popdensity  popwhite  popblack  \
0     561      ADAMS    IL  0.052     66090  1270.961540     63917      1702   
1     562  ALEXANDER    IL  0.014     10626   759.000000      7054      3496   
2     563       BOND    IL  0.022     14991   681.409091     14477       429   
3     564      BOONE    IL  0.017     30806  1812.117650     29344       127   
4     565      BROWN    IL  0.018      5836   324.222222      5264       547   
..    ...        ...   ...    ...       ...          ...       ...       ...   
432  3048   WAUKESHA    WI  0.034    304715  8962.205880    298313      1096   
433  3049    WAUPACA    WI  0.045     46104  1024.533330     45695        22   
434  3050   WAUSHARA    WI  0.037     19385   523.918919     19094        29   
435  3051  WINNEBAGO    WI  0.035    140320  4009.142860    136822       697   
436  3052       WOOD    WI  0.048     73605  1533.437500     72157        90   

     popamerindian  popasian  ...  percollege  percprof  poppovertyknown  \
0               98       249  ...   19.631392  4.355859            63628   
1               19        48  ...   11.243308  2.870315            10529   
2               35        16  ...   17.033819  4.488572            14235   
3               46       150  ...   17.278954  4.197800            30337   
4               14         5  ...   14.475999  3.367680             4815   
..             ...       ...  ...         ...       ...              ...   
432            672      2699  ...   35.396784  7.667090           299802   
433            125        92  ...   16.549869  3.138596            44412   
434             70        43  ...   15.064584  2.620907            19163   
435            685      1728  ...   24.995504  5.659847           133950   
436            481       722  ...   21.666382  4.583725            72685   

     percpovertyknown  percbelowpoverty  percchildbelowpovert  \
0           96.274777         13.151443             18.011717   
1           99.087145         32.244278             45.826514   
2           94.956974         12.068844             14.036061   
3           98.477569          7.209019             11.179536   
4           82.505140         13.520249             13.022889   
..                ...               ...                   ...   
432         98.387674          3.121060              3.785820   
433         96.330036          8.488697             10.071411   
434         98.854785         13.786985             20.050708   
435         95.460376          8.804031             10.592031   
436         98.750085          8.525831             11.162997   

     percadultpoverty  percelderlypoverty  inmetro  category  
0           11.009776           12.443812        0       AAR  
1           27.385647           25.228976        0       LHR  
2           10.852090           12.697410        0       AAR  
3            5.536013            6.217047        1       ALU  
4           11.143211           19.200000        0       AAR  
..                ...                 ...      ...       ...  
432          2.590061            4.085479        1       HLU  
433          6.953799           10.338641        0       AAR  
434         11.695784           11.804558        0       AAR  
435          8.660587            6.661094        1       HAU  
436          7.375656            7.882918        0       AAR  

[437 rows x 28 columns]

## 명령어 사용하여 데이터 특징 파악하기 ##
메서드: 변수가 지니고 있는 함수로, 변수 뒤에 점을 찍어 사용한다. 변수의 자료 구조에 따라 사용가능 메서드가 다르다.

### head(), tail()로 앞뒤 행의 데이터 출력 ###


In [6]:
df_midwest.head()

PID     county state   area  poptotal   popdensity  popwhite  popblack  \
0  561      ADAMS    IL  0.052     66090  1270.961540     63917      1702   
1  562  ALEXANDER    IL  0.014     10626   759.000000      7054      3496   
2  563       BOND    IL  0.022     14991   681.409091     14477       429   
3  564      BOONE    IL  0.017     30806  1812.117650     29344       127   
4  565      BROWN    IL  0.018      5836   324.222222      5264       547   

   popamerindian  popasian  ...  percollege  percprof  poppovertyknown  \
0             98       249  ...   19.631392  4.355859            63628   
1             19        48  ...   11.243308  2.870315            10529   
2             35        16  ...   17.033819  4.488572            14235   
3             46       150  ...   17.278954  4.197800            30337   
4             14         5  ...   14.475999  3.367680             4815   

   percpovertyknown  percbelowpoverty  percchildbelowpovert  percadultpoverty  \
0         96.274777         13.151443             18.011717         11.009776   
1         99.087145         32.244278             45.826514         27.385647   
2         94.956974         12.068844             14.036061         10.852090   
3         98.477569          7.209019             11.179536          5.536013   
4         82.505140         13.520249             13.022889         11.143211   

   percelderlypoverty  inmetro  category  
0           12.443812        0       AAR  
1           25.228976        0       LHR  
2           12.697410        0       AAR  
3            6.217047        1       ALU  
4           19.200000        0       AAR  

[5 rows x 28 columns]

In [12]:
df_midwest.head(10)

PID     county state   area  poptotal   popdensity  popwhite  popblack  \
0  561      ADAMS    IL  0.052     66090  1270.961540     63917      1702   
1  562  ALEXANDER    IL  0.014     10626   759.000000      7054      3496   
2  563       BOND    IL  0.022     14991   681.409091     14477       429   
3  564      BOONE    IL  0.017     30806  1812.117650     29344       127   
4  565      BROWN    IL  0.018      5836   324.222222      5264       547   
5  566     BUREAU    IL  0.050     35688   713.760000     35157        50   
6  567    CALHOUN    IL  0.017      5322   313.058824      5298         1   
7  568    CARROLL    IL  0.027     16805   622.407407     16519       111   
8  569       CASS    IL  0.024     13437   559.875000     13384        16   
9  570  CHAMPAIGN    IL  0.058    173025  2983.189660    146506     16559   

   popamerindian  popasian  ...  percollege   percprof  poppovertyknown  \
0             98       249  ...   19.631392   4.355859            63628   
1             19        48  ...   11.243308   2.870315            10529   
2             35        16  ...   17.033819   4.488572            14235   
3             46       150  ...   17.278954   4.197800            30337   
4             14         5  ...   14.475999   3.367680             4815   
5             65       195  ...   18.904624   3.275891            35107   
6              8        15  ...   11.917388   3.209601             5241   
7             30        61  ...   16.197121   3.055727            16455   
8              8        23  ...   14.107649   3.206799            13081   
9            331      8033  ...   41.295808  17.757448           154934   

   percpovertyknown  percbelowpoverty  percchildbelowpovert  percadultpoverty  \
0         96.274777         13.151443             18.011717         11.009776   
1         99.087145         32.244278             45.826514         27.385647   
2         94.956974         12.068844             14.036061         10.852090   
3         98.477569          7.209019             11.179536          5.536013   
4         82.505140         13.520249             13.022889         11.143211   
5         98.372002         10.399635             14.158819          8.179287   
6         98.478016         15.149781             13.787761         12.932331   
7         97.917287         11.710726             17.225462         10.027037   
8         97.350599         13.875086             17.994784         11.914343   
9         89.544286         15.572437             14.132234         17.562728   

   percelderlypoverty  inmetro  category  
0           12.443812        0       AAR  
1           25.228976        0       LHR  
2           12.697410        0       AAR  
3            6.217047        1       ALU  
4           19.200000        0       AAR  
5           11.008586        0       AAR  
6           21.085271        0       LAR  
7            9.525052        0       AAR  
8           13.660180        0       AAR  
9            8.105017        1       HAU  

[10 rows x 28 columns]

In [7]:
df_midwest.tail()

PID     county state   area  poptotal   popdensity  popwhite  popblack  \
432  3048   WAUKESHA    WI  0.034    304715  8962.205880    298313      1096   
433  3049    WAUPACA    WI  0.045     46104  1024.533330     45695        22   
434  3050   WAUSHARA    WI  0.037     19385   523.918919     19094        29   
435  3051  WINNEBAGO    WI  0.035    140320  4009.142860    136822       697   
436  3052       WOOD    WI  0.048     73605  1533.437500     72157        90   

     popamerindian  popasian  ...  percollege  percprof  poppovertyknown  \
432            672      2699  ...   35.396784  7.667090           299802   
433            125        92  ...   16.549869  3.138596            44412   
434             70        43  ...   15.064584  2.620907            19163   
435            685      1728  ...   24.995504  5.659847           133950   
436            481       722  ...   21.666382  4.583725            72685   

     percpovertyknown  percbelowpoverty  percchildbelowpovert  \
432         98.387674          3.121060              3.785820   
433         96.330036          8.488697             10.071411   
434         98.854785         13.786985             20.050708   
435         95.460376          8.804031             10.592031   
436         98.750085          8.525831             11.162997   

     percadultpoverty  percelderlypoverty  inmetro  category  
432          2.590061            4.085479        1       HLU  
433          6.953799           10.338641        0       AAR  
434         11.695784           11.804558        0       AAR  
435          8.660587            6.661094        1       HAU  
436          7.375656            7.882918        0       AAR  

[5 rows x 28 columns]

In [13]:
df_midwest.tail(10)

PID      county state   area  poptotal   popdensity  popwhite  popblack  \
427  3043      VERNON    WI  0.048     25617   533.687500     25509        12   
428  3044       VILAS    WI  0.060     17707   295.116667     16116         9   
429  3045    WALWORTH    WI  0.032     75000  2343.750000     72747       454   
430  3046    WASHBURN    WI  0.050     13772   275.440000     13585        25   
431  3047  WASHINGTON    WI  0.025     95328  3813.120000     94465       125   
432  3048    WAUKESHA    WI  0.034    304715  8962.205880    298313      1096   
433  3049     WAUPACA    WI  0.045     46104  1024.533330     45695        22   
434  3050    WAUSHARA    WI  0.037     19385   523.918919     19094        29   
435  3051   WINNEBAGO    WI  0.035    140320  4009.142860    136822       697   
436  3052        WOOD    WI  0.048     73605  1533.437500     72157        90   

     popamerindian  popasian  ...  percollege  percprof  poppovertyknown  \
427             36        42  ...   18.942131  3.624948            25087   
428           1534        38  ...   19.211861  4.315256            17446   
429            201       494  ...   23.156904  6.082324            71553   
430            122        33  ...   19.016887  4.022803            13532   
431            208       337  ...   23.390900  4.014568            94143   
432            672      2699  ...   35.396784  7.667090           299802   
433            125        92  ...   16.549869  3.138596            44412   
434             70        43  ...   15.064584  2.620907            19163   
435            685      1728  ...   24.995504  5.659847           133950   
436            481       722  ...   21.666382  4.583725            72685   

     percpovertyknown  percbelowpoverty  percchildbelowpovert  \
427         97.931061         15.824929             21.201105   
428         98.526007         14.736902             23.043367   
429         95.404000          9.641804              8.699613   
430         98.257334         15.866095             21.418598   
431         98.756923          3.237628              4.069854   
432         98.387674          3.121060              3.785820   
433         96.330036          8.488697             10.071411   
434         98.854785         13.786985             20.050708   
435         95.460376          8.804031             10.592031   
436         98.750085          8.525831             11.162997   

     percadultpoverty  percelderlypoverty  inmetro  category  
427         13.449643           14.571429        0       AAR  
428         14.251978            9.173228        0       AAR  
429         10.926610            6.894182        0       AAR  
430         13.642483           14.329455        0       AAR  
431          2.584500            4.280889        1       HLU  
432          2.590061            4.085479        1       HLU  
433          6.953799           10.338641        0       AAR  
434         11.695784           11.804558        0       AAR  
435          8.660587            6.661094        1       HAU  
436          7.375656            7.882918        0       AAR  

[10 rows x 28 columns]

### shape로 데이터프레임의 행열 개수 출력 ###
shape는 함수가 아닌 속성(attribute)이므로 괄호를 입력하지 않는다!

In [8]:
df_midwest.shape

(437, 28)

*-> df_midwest가 437개의 행, 28개의 열을 가지고 있음을 확인할 수 있다.*

### info()로 모든 변수의 속성 한눈에 파악하기 ###
info()에도 행열 개수 정보가 출력된다. 한눈에 파악하기 위해 shape보다 더 자주 사용되는 명령어어

In [9]:
df_midwest.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 437 entries, 0 to 436
Data columns (total 28 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   PID                   437 non-null    int64  
 1   county                437 non-null    object 
 2   state                 437 non-null    object 
 3   area                  437 non-null    float64
 4   poptotal              437 non-null    int64  
 5   popdensity            437 non-null    float64
 6   popwhite              437 non-null    int64  
 7   popblack              437 non-null    int64  
 8   popamerindian         437 non-null    int64  
 9   popasian              437 non-null    int64  
 10  popother              437 non-null    int64  
 11  percwhite             437 non-null    float64
 12  percblack             437 non-null    float64
 13  percamerindan         437 non-null    float64
 14  percasian             437 non-null    float64
 15  percother             4


- *첫째 줄에서 df_midwest가 padas로 만들어진 데이터프레임이라는 것을 확인*
- *둘째 줄에서 437개의 행으로 이루어졌으며, 행 번호가 0부터 436까지라는 것을 확인 (cf. 파이썬의 셈은 0부터 시작)*
- *셋째 줄에서 df_midwest가 변수(열) 28개로 구성된다는 것을 확인*
- *이하에서 df_midwest의 변수 속성을 확인 (county, state, category를 제외하면 전부 float/int 계량형)*

### describe()로 변수 값의 요약통계량 확인하기 ###

In [10]:
df_midwest.describe()

PID        area      poptotal    popdensity      popwhite  \
count   437.000000  437.000000  4.370000e+02    437.000000  4.370000e+02   
mean   1437.338673    0.033169  9.613030e+04   3097.742985  8.183992e+04   
std     876.390266    0.014679  2.981705e+05   7664.751786  2.001966e+05   
min     561.000000    0.005000  1.701000e+03     85.050000  4.160000e+02   
25%     670.000000    0.024000  1.884000e+04    622.407407  1.863000e+04   
50%    1221.000000    0.030000  3.532400e+04   1156.208330  3.447100e+04   
75%    2059.000000    0.038000  7.565100e+04   2330.000000  7.296800e+04   
max    3052.000000    0.110000  5.105067e+06  88018.396600  3.204947e+06   

           popblack  popamerindian       popasian       popother   percwhite  \
count  4.370000e+02     437.000000     437.000000     437.000000  437.000000   
mean   1.102388e+04     343.109840    1310.464531    1612.931350   95.558441   
std    7.895827e+04     868.926751    9518.394189   18526.540699    7.087358   
min    0.000000e+00       4.000000       0.000000       0.000000   10.694087   
25%    2.900000e+01      44.000000      35.000000      20.000000   94.886032   
50%    2.010000e+02      94.000000     102.000000      66.000000   98.032742   
75%    1.291000e+03     288.000000     401.000000     345.000000   99.074935   
max    1.317147e+06   10289.000000  188565.000000  384119.000000   99.822821   

       ...     perchsd  percollege    percprof  poppovertyknown  \
count  ...  437.000000  437.000000  437.000000     4.370000e+02   
mean   ...   73.965546   18.272736    4.447259     9.364228e+04   
std    ...    5.843177    6.261908    2.408427     2.932351e+05   
min    ...   46.912261    7.336108    0.520291     1.696000e+03   
25%    ...   71.325329   14.113725    2.997957     1.836400e+04   
50%    ...   74.246891   16.797562    3.814239     3.378800e+04   
75%    ...   77.195345   20.549893    4.949324     7.284000e+04   
max    ...   88.898674   48.078510   20.791321     5.023523e+06   

       percpovertyknown  percbelowpoverty  percchildbelowpovert  \
count        437.000000        437.000000            437.000000   
mean          97.110267         12.510505             16.447464   
std            2.749863          5.150155              7.228634   
min           80.902441          2.180168              1.918955   
25%           96.894572          9.198715             11.624088   
50%           98.169562         11.822313             15.270164   
75%           98.598636         15.133226             20.351878   
max           99.860384         48.691099             64.308477   

       percadultpoverty  percelderlypoverty     inmetro  
count        437.000000          437.000000  437.000000  
mean          10.918798           11.389043    0.343249  
std            5.109166            3.661259    0.475338  
min            1.938504            3.547067    0.000000  
25%            7.668009            8.911763    0.000000  
50%           10.007610           10.869119    0.000000  
75%           13.182182           13.412162    1.000000  
max           43.312464           31.161972    1.000000  

[8 rows x 25 columns]

In [11]:
df_midwest.describe(include= 'all')

PID   county state        area      poptotal    popdensity  \
count    437.000000      437   437  437.000000  4.370000e+02    437.000000   
unique          NaN      320     5         NaN           NaN           NaN   
top             NaN  JACKSON    IL         NaN           NaN           NaN   
freq            NaN        5   102         NaN           NaN           NaN   
mean    1437.338673      NaN   NaN    0.033169  9.613030e+04   3097.742985   
std      876.390266      NaN   NaN    0.014679  2.981705e+05   7664.751786   
min      561.000000      NaN   NaN    0.005000  1.701000e+03     85.050000   
25%      670.000000      NaN   NaN    0.024000  1.884000e+04    622.407407   
50%     1221.000000      NaN   NaN    0.030000  3.532400e+04   1156.208330   
75%     2059.000000      NaN   NaN    0.038000  7.565100e+04   2330.000000   
max     3052.000000      NaN   NaN    0.110000  5.105067e+06  88018.396600   

            popwhite      popblack  popamerindian       popasian  ...  \
count   4.370000e+02  4.370000e+02     437.000000     437.000000  ...   
unique           NaN           NaN            NaN            NaN  ...   
top              NaN           NaN            NaN            NaN  ...   
freq             NaN           NaN            NaN            NaN  ...   
mean    8.183992e+04  1.102388e+04     343.109840    1310.464531  ...   
std     2.001966e+05  7.895827e+04     868.926751    9518.394189  ...   
min     4.160000e+02  0.000000e+00       4.000000       0.000000  ...   
25%     1.863000e+04  2.900000e+01      44.000000      35.000000  ...   
50%     3.447100e+04  2.010000e+02      94.000000     102.000000  ...   
75%     7.296800e+04  1.291000e+03     288.000000     401.000000  ...   
max     3.204947e+06  1.317147e+06   10289.000000  188565.000000  ...   

        percollege    percprof  poppovertyknown  percpovertyknown  \
count   437.000000  437.000000     4.370000e+02        437.000000   
unique         NaN         NaN              NaN               NaN   
top            NaN         NaN              NaN               NaN   
freq           NaN         NaN              NaN               NaN   
mean     18.272736    4.447259     9.364228e+04         97.110267   
std       6.261908    2.408427     2.932351e+05          2.749863   
min       7.336108    0.520291     1.696000e+03         80.902441   
25%      14.113725    2.997957     1.836400e+04         96.894572   
50%      16.797562    3.814239     3.378800e+04         98.169562   
75%      20.549893    4.949324     7.284000e+04         98.598636   
max      48.078510   20.791321     5.023523e+06         99.860384   

        percbelowpoverty  percchildbelowpovert  percadultpoverty  \
count         437.000000            437.000000        437.000000   
unique               NaN                   NaN               NaN   
top                  NaN                   NaN               NaN   
freq                 NaN                   NaN               NaN   
mean           12.510505             16.447464         10.918798   
std             5.150155              7.228634          5.109166   
min             2.180168              1.918955          1.938504   
25%             9.198715             11.624088          7.668009   
50%            11.822313             15.270164         10.007610   
75%            15.133226             20.351878         13.182182   
max            48.691099             64.308477         43.312464   

        percelderlypoverty     inmetro  category  
count           437.000000  437.000000       437  
unique                 NaN         NaN        16  
top                    NaN         NaN       AAR  
freq                   NaN         NaN       193  
mean             11.389043    0.343249       NaN  
std               3.661259    0.475338       NaN  
min               3.547067    0.000000       NaN  
25%               8.911763    0.000000       NaN  
50%              10.869119    0.000000       NaN  
75%              13.412162    1.000000       N

- *문자인 변수의 요약 통계량을 출력하려면 .describe(include = 'all')*
- *unique, top, freq는 문자 변수로만 계산. 숫자 변수에는 NaN*